src/GafferArnold/InteractiveArnoldRender.cpp

```diff
@@ -81,29 +79,26 @@ InteractiveArnoldRender::~InteractiveArnoldRender()
 
 void InteractiveArnoldRender::flushCaches( int flags )
 {
-	std::vector<Interrupted> interrupted;
-
-	const InstanceSet &i = instances();
-	for( InstanceSet::const_iterator it = i.begin(), eIt = i.end(); it != eIt; ++it )
+	/// \todo Perhaps this makes more sense as a non-static method
+	/// that can be called directly on individual instances?
+	for( const auto &instance : instances() )
 	{
-		IntPlug *statePlug = (*it)->statePlug()->source<IntPlug>();
+		IntPlug *statePlug = instance->statePlug()->source<IntPlug>();
 		if( !statePlug->settable() )
 		{
 			continue;
 		}
 
-		const State state = (InteractiveRender::State)statePlug->getValue();
-		if( state != Stopped )
+		if( !instance->renderer() )
 		{
-			statePlug->setValue( Stopped );
-			interrupted.push_back( Interrupted( statePlug, state ) );
+			continue;
 		}
-	}
 
-	AiUniverseCacheFlush( flags );
+		const State state = (InteractiveRender::State)statePlug->getValue();
+		statePlug->setValue( Paused );
+
+		instance->renderer()->command( "ai:cacheFlush", { { "flags", new IECore::IntData( flags ) } } );
```

Comment:
I'm a bit skeptical of whether we're gaining much through introducing this new `command( "ai:cacheFlush" )` API - we have to know that we're dealing with Arnold anyway in order to know to call ai:cacheFlush, so would there really be any disadvantage to just having ArnoldRenderer expose a way of getting the AtUniverse, so we could just call AiUniverseCacheFlush here?

I don't mind this, but it feels like a little bit of indirection without accomplishing much.

Reply:
Yeah, I wobbled back and forth a bit on this before ending up with the command. I don't think there's a huge benefit to it right now, but it does seem a bit cleaner to avoid direct Arnold API access and allow the renderer interface to take responsibility for all of that. If it involved a lot more boilerplate it might have tipped the balance, but adding the command was just a handful of lines of code.

Slightly related, I think there's a decent case for dropping ArnoldRender/InteractiveArnoldRender in future, and just using the Render/InteractiveRender nodes directly. Either with plugs on the node to choose the renderer, or perhaps even a scene option. At that point, better encapsulation via commands might have more benefit.